OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseDiscussionTopicsFragment.java

```diff
@@ -115,17 +99,29 @@ public boolean onQueryTextChange(String newText) {
                 return false;
             }
         });
-
-        binding.discussionTopicsListview.setOnItemClickListener(
-                (parent, view1, position, id) -> router.showCourseDiscussionPostsForDiscussionTopic(
-                        requireActivity(),
-                        ((DiscussionTopicDepth) parent.getItemAtPosition(position)).getDiscussionTopic(),
-                        courseData));
-
         getTopicList();
         showCourseDiscussionTopic();
     }
 
+    private void addDiscussionHeaders() {
```

Comment:
This function should return the headers only and we should do the other operations i.e clear, add somewhere else.

Rename it to -> `getDdiscussionHeaders()`

Reply:
I don't think so as we need to clear the list first then headers will be added, it is a helping method not a return-type method. if we make it a return type method then we need to create another method to check n clear the list before adding the headers.

Reply:
IMO, we can add doc for the method for better understanding